### ISSUE TITLE
Helm3: "hook-delete-policy : hook-succeeded" is ignored

### ISSUE THREAD

**Issue by alitari** [helm/helm#5797]
Output of `helm version`: version.BuildInfo{Version:"v3.0.0-alpha.1

Output of `kubectl version`: Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.11"

Cloud Provider/Platform (AKS, GKE, Minikube etc.): Docker-for-desktop (kubernetes enabled)

The [helm hook feature "hook-delete-policy : hook-succeeded"](https://helm.sh/docs/charts_hooks/#writing-a-hook) which deletes resources after successful deployment does not work, the resources remain in the cluster.

e.g. pre-install-cm.yaml
```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: hook-test-preinstall
  labels:
    app: hook-test
    chart: hook-test
  annotations:
    "helm.sh/hook": pre-install,pre-upgrade
    "helm.sh/hook-weight": "-10"
    "helm.sh/hook-delete-policy": hook-succeeded
data:
  hello: world should not be there
```

**Comment by MaikuMori** [comment]
Can confirm that it doesn't work. Rather annoying :(

**Comment by ericabr** [comment]
I can replicate with an earlier version of helm. `v2.12.3`

**Comment by unguiculus** [comment]
Should be fixed in #6054
